content/articles/visualizing-repositories-using-pygal/index.md

```diff
@@ -0,0 +1,371 @@
+Pygal allows us to create a range of graphs and charts. You'll learn how to construct a visualisation that shows the comparative popularity of Python works on GitHub in this tutorial. We'll create an interactive bar chart, where the height of each bar will represent the sum of stars earned by the project. By clicking a bar, you'll be sent to the project's GitHub page.
+
+### Prerequisites
+
+You'll need some Python skills to follow along with this tutorial. You must be in a position to work with web APIs in Python.
+
+### Installing pygal
+
+Consider the following command to install pygal:
+
+```python
+
+$ pip install pygal
+
+```
+
+Visit [this](http://www.pygal.org/en/stable/installing.html) if this is your first time using pip.
+
+### The pygal gallery
+
+Check the chart types page to explore what types of charts Pygal can generate: Select **Documentation**, finally **Chart types** in [pygal](http://www.pygal.org/). You can examine how the visualizations are created by looking at the source code for each case.
+
+### Visualizing repositories
+
+Since we'll want to incorporate more than one repository in our visualisation. Let's create a loop that prints specified information about each of the repositories supplied by the API call so that we can add them all to the visualization:
+
+```python
+
+import requests
+
+url = 'https://api.github.com/search/repositories?q=language:python&sort=stars' # Initiate an API request.
+
+request = requests.get(url)
+
+print("Status code:", request.status_code)
+
+respons_dict = request.json()  # In a variable, save the API response.
+
+print("Total repos:", respons_dict['total_count'])
+
+# Examine the repository's details.
+
+repos_dicts = respons_dict['items']
+
+print("Repos found:", len(repos_dicts))
+
+print("\nSelected info about each repository:")
+
+for repos_dict in repos_dicts:   #go through all the dictionaries in repos_dicts.
+
+    print('name-:', repos_dict['name'])
+
+    print('owner-:', repos_dict['owner']['login'])
+
+    print('stars-:', repos_dict['stargazers_count'])
+
+    print('repo-:', repos_dict['html_url'])
+
+    print('Description-:', repos_dict['description'])
+
+```
```

Comment:
You can add a little more comments where necessary within the article, for even beginners to flow easily.
Also use more connector words like 'next or second' to keep the article structured